compiler/rustc_const_eval/src/const_eval/mod.rs

```diff
@@ -90,19 +115,67 @@ fn const_to_valtree_inner<'tcx>(
         // Technically we could allow function pointers (represented as `ty::Instance`), but this is not guaranteed to
         // agree with runtime equality tests.
         ty::FnPtr(_) | ty::RawPtr(_) => None,
-        ty::Ref(..) => unimplemented!("need to use deref_const"),
+
+        ty::Ref(_, inner_ty, _)  => {
+            match inner_ty.kind() {
+                ty::Slice(_) | ty::Str => {
+                    match ecx.try_read_immediate_from_mplace(&place) {
```

Comment:
Why don't you use `read_immediate`? A reference should *always* be an immediate.

There's a reason `try_read_immediate_from_mplace` was private, it should not usually be used. :)